app/tests/unit/forms/form/exportService.spec.js

```diff
@@ -247,6 +247,94 @@ describe('_buildCsvHeaders', () => {
   });
 });
 
+describe('_buildCsvHeaders', () => {
+  it('should build correct csv headers for single row export option when selected fields provided', async () => {
+    // form schema from db
+    const formSchema = require('../../../fixtures/form/Kitchen_sink_form_schema_datagrid.json');
+
+    // form object from db
+    const form = { id: 123 };
+
+    // mock latestFormSchema
+    exportService._readLatestFormSchema = jest.fn(() => {
+      return formSchema;
+    });
+
+    // submissions export (matchces the format that is downloaded in UI)
+    const submissionsExport = require('../../../fixtures/submission/kitchen_sink_submission_data_export_datagrid.json');
+
+    // build csv headers
+    // get fields which we gonna filtered column needed from
+    const fields = require('../../../fixtures/submission/kitchen_sink_submission_data_export_datagrid_fields_selection.json');
+    // get result columns if we need to filter out the columns
+    const result = await exportService._buildCsvHeaders(form, submissionsExport, 1, fields, true);
+
+    expect(result).toHaveLength(20);
+    expect(result).toEqual(
+      expect.arrayContaining([
+        'form.confirmationId',
+        'oneRowPerLake.0.closestTown',
+        'oneRowPerLake.0.dataGrid.0.fishType',
+        'oneRowPerLake.0.dataGrid.1.fishType',
+        'oneRowPerLake.0.dataGrid.0.numberKept',
+        'oneRowPerLake.0.dataGrid.1.fishType',
```

Comment:
Looks like `'oneRowPerLake.0.dataGrid.1.fishType'` is repeated, is this correct? Or is it

```
        'oneRowPerLake.0.dataGrid.1.numberKept',
```